core/new-gui/src/app/workspace/service/workflow-graph/model/workflow-action.service.ts

```diff
@@ -226,7 +226,7 @@ export class WorkflowActionService {
   public addPort(operatorID: string, isInput: boolean, allowMultiInputs?: boolean): void {
     const operator = this.texeraGraph.getOperator(operatorID);
     const prefix = isInput ? "input-" : "output-";
-    let suffix = isInput ? operator.inputPorts.length + 1 : operator.outputPorts.length + 1;
+    let suffix = isInput ? operator.inputPorts.length : operator.outputPorts.length;
```

Comment:
just curious, in the current version, what happens if two users clicks the add port button at the same time (in different browsers)? it seems the ID (suffix) is generated locally and will be synced by the shared editing server later.

Reply:
This part of the code is not written by me, but most probably it will be a conflict (the two ports will have the same portID.) I changed the port ID part to be a uuid, and this `suffix` part of code to be only used for port display name (since uuid is ugly as a name). So port display name has a slight chance to be the same, but port id will not.

Reply:
will add to description about this limitation